frontend/src/components/login-form/index.spec.tsx

```diff
@@ -0,0 +1,272 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+import Router, { useRouter } from "next/router";
+import { mocked } from "ts-jest/utils";
+
+import {
+  act,
+  fireEvent,
+  MockedProvider,
+  render,
+  screen,
+  wait,
+} from "~/test-utils";
+import { LoginDocument } from "~/types";
+
+import { LOGIN_KEY } from "../profile/hooks";
+import { LoginForm } from "./index";
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+jest.mock("next/router", () => ({
+  push: jest.fn(),
+  useRouter: jest.fn(() => ({
+    prefetch: () => null,
+    events: {
+      on: () => {},
+      off: () => {},
+    },
+    replace: jest.fn().mockResolvedValue(true),
+    pathname: "",
+    route: "",
+    query: {},
+  })),
+}));
+
+const MockedRouterPush = mocked(Router.push, true);
+const MockedUseRouter = mocked(useRouter, true);
+
+const createLoginMock = (
+  data,
+  input = {
+    email: "email@email.it",
+    password: "password",
+  },
+) => [
+  {
+    request: {
+      query: LoginDocument,
+      variables: {
+        input,
+      },
+    },
+    result: {
+      data,
+    },
+  },
+];
+
+const VALID_LOGIN_MOCKS = createLoginMock({
+  login: {
+    __typename: "LoginSuccess",
+  },
+});
+
+describe("Login form", () => {
+  test("should render", () => {
+    render(
+      <MockedProvider>
+        <LoginForm />
+      </MockedProvider>,
+    );
+
+    expect(screen.getByText("Don't have an account?")).toBeInTheDocument();
```

Comment:
might be best to use ids :D

Reply:
https://testing-library.com/docs/guiding-principles/

https://testing-library.com/docs/queries/bytestid/

Testing by ID is not recommended as the tests should test what the user sees as far I understood

Reply:
> That said, they are way better than querying based on DOM structure or styling css class names

It's not saying to use text :D I don't really like the idea of hardcoding the labels in the tests :)

Reply:
Test ID is the only alternative to match it I think then, because css classes are randomly generated

Reply:
> > That said, they are way better than querying based on DOM structure or styling css class names
> 
> It's not saying to use text :D I don't really like the idea of hardcoding the labels in the tests :)

I am also not sure about this, I think querying labels/text is quite common in the FE. I will have a look in a bit and see what others do